### ISSUE TITLE
custom optional annotation with Lombok not honored

### ISSUE THREAD

**Issue by niels1voo** [vojtechhabarta/typescript-generator#202]
Hi, I discovered that the behavior of the plugin has changed since 1.25.322 with respect to Lombok processed classes (we were using jackson 1 before).  Please see my fork https://github.com/niels1voo/typescript-generator, where I have modified the sample-maven module to demonstrate the issues with version 1.30.381 w/jackson2

What I have found:

1. custom optional annotations are disregarded when applied in a Lombok-annotated class (@Data). 
   - this used to work in 1.25.322
2. properties defined following a property with the @Transient annotation are not generated at all.

**Comment by vojtechhabarta** [comment]
Hi @niels1voo,
thanks for the report with sample.

Version [1.29.355](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v1.29.355) had changes that affected optional properties.

### Point 2
Previously typescript-generator loaded all Jackson2 modules automatically. This caused problems so since v1.29.355 this is no longer the case. `@javax.persistence.Transient` is handled by `Hibernate5Module` from `jackson-datatype-hibernate5`. You can add this module like this:
``` xml
<jackson2Modules>
    <module>com.fasterxml.jackson.datatype.hibernate5.Hibernate5Module</module>
</jackson2Modules>
```
or you can set `jackson2ModuleDiscovery` to `true`. See [Breaking changes](https://github.com/vojtechhabarta/typescript-generator/wiki/Breaking-Changes).

### Point 1
This was unintended breaking change: previous versions used `BeanPropertyWriter.getAnnotation()`
 for reading annotations and in v1.29.355 I changed this to reading annotations directly from the class member. Apparently this is not equivalent so in 57c9d9f54e65f265f19ab86ff5a771b1e0a1d850 (in master branch) I changed it back to `BeanPropertyWriter.getAnnotation()`. Could you please try it?

**Comment by niels1voo** [comment]
57c9d9f fixed the custom annotation issue in 1.

**Comment by vojtechhabarta** [comment]
Released [v1.30.383](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v1.30.383).
